lib/cucumber/core/test/runner.rb

```diff
@@ -4,40 +4,39 @@ module Cucumber
   module Core
     module Test
       class Runner
-        attr_reader :report, :running_test_case, :running_test_step
-        private :report, :running_test_case, :running_test_step
+        attr_reader :events, :running_test_case, :running_test_step
+        private :events, :running_test_case, :running_test_step
 
-        def initialize(report)
-          @report = report
+        def initialize(events)
+          @events = events
         end
 
         def test_case(test_case, &descend)
           @running_test_case = RunningTestCase.new
           @running_test_step = nil
-          report.before_test_case(test_case)
+          events.test_case_starting(test_case)
           descend.call(self)
```

Comment:
This question is more for my own edification, but why `descend.call(self)` instead of `yield self`? I don't know what the difference is.